core/src/main/java/org/apache/iceberg/rest/requests/FetchScanTasksRequest.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.rest.requests;
+
+import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.rest.RESTRequest;
+
+public class FetchScanTasksRequest implements RESTRequest {
+
+  private String planTask;
+
+  public FetchScanTasksRequest(String planTask) {
+    this.planTask = planTask;
+    validate();
```

Comment:
we typically don't call `validate` in the constructor

Reply:
Just curious why we dont call `validate` in the constructor? If a faulty request or response is being constructed it would catch this before serial/deserial time. 

cc @amogh-jahagirdar

Reply:
It seems this pattern is done here as well: https://github.com/apache/iceberg/blob/main/core/src/main/java/org/apache/iceberg/rest/responses/ConfigResponse.java#L64